river/tree/mondrian/mondrian_tree_nodes.py

```diff
@@ -0,0 +1,474 @@
+import typing
+
+from abc import ABC, abstractmethod
+
+from river.utils.math import log_sum_2_exp
+
+from math import log
+
+from river.tree.base import Leaf
+from river.tree.base import Branch
+
+
+class MondrianTreeLeaf(Leaf, ABC):
+    """
+    Abstract class for all types of nodes in a Mondrian Tree
+
+    Parameters
+    ----------
+    parent
+        Parent Node
+    n_features
+        Number of features of the input data
+    time
+        Split time of the node for Mondrian process
+    """
+
+    def __init__(
+            self,
+            parent: Leaf or None,
+            n_features: int,
+            time: float
+    ):
+
+        super().__init__()
+
+        # Generic Node attributes
+        self.parent = parent
+        self.time = time
+        self.is_leaf = True
+        self.depth = 0
```

Comment:
Does this change?

Reply:
What are you targeting exactly ? I believe depth can change even after initialization, I'm not sure about the rest, but possibly it does (see when the "copy" function is being used, we push a node information into another).
I'd have to check it into the IDE though, I'm just throwing that here =)

Reply:
Hi ! Looking through the code right now, here is what I found out:
- parent: parent of the node can change
- time: splitting time of the node can change
- is_leaf: it does change for all node that is splitting
- depth: it is updated